pydoctor/sphinx_ext/api_output.py

```diff
@@ -0,0 +1,69 @@
+"""
+Generate the API docs using pydoctor to be integrated into Sphinx build system.
+
+Inside the Sphinx conf.py file you need to define the following configuration options:
+
+* pydoctor_cwd: Absolute path in which the pydoctor build is trigerred.
+* pydoctor_args: A list with all the pydoctor command line arguments used to trigger the build.
+
+You must call pydoctor with `--quiet` argument
+as otherwise any extra output is converted into Sphinx warnings.
+"""
+import os
+from sphinx.util import logging
+from sphinx.errors import ConfigError
+
+from contextlib import redirect_stdout
+from io import StringIO
+from pydoctor import __version__
+from pydoctor.driver import main
+
+logger = logging.getLogger(__name__)
+
+
+def on_build_finished(app, exception):
+    """
+    Called when Sphinx build is done.
+    """
+    if not app.config.pydoctor_args:
+        raise ConfigError("Missing 'pydoctor_args'.")
+
+    if not app.config.pydoctor_cwd:
+        raise ConfigError("Missing 'pydoctor_cwd'.")
+
+    placeholders = {
+        'outdir': app.outdir,
+        }
+
+    args = []
+    for argument in app.config.pydoctor_args:
+        args.append(argument.format(**placeholders))
+
+    stream = StringIO()
+
+    old_cwd = os.getcwd()
+    try:
+        os.chdir(app.config.pydoctor_cwd)
```

Comment:
Do we need to change the working dir? If passed absolute paths, pydoctor should be able to run from any directory, right?

Reply:
We don't need it. Great call. Thanks!